sokol_app.h

```diff
@@ -5005,24 +5004,31 @@ EM_JS(void, sapp_js_fetch_dropped_file, (int index, _sapp_html5_fetch_callback c
     reader.readAsArrayBuffer(files[index]);
 });
 
-EM_JS(void, sapp_js_remove_dragndrop_listeners, (const char* canvas_name_cstr), {
-    const canvas_name = UTF8ToString(canvas_name_cstr);
-    const canvas = document.getElementById(canvas_name);
+EM_JS(void, sapp_js_remove_dragndrop_listeners, (const char* canvas_selector_cstr), {
+    const canvas = Modules.canvas;
     canvas.removeEventListener('dragenter', Module.sokol_dragenter);
     canvas.removeEventListener('dragleave', Module.sokol_dragleave);
     canvas.removeEventListener('dragover',  Module.sokol_dragover);
     canvas.removeEventListener('drop',      Module.sokol_drop);
 });
 
-EM_JS(void, sapp_js_init, (const char* c_str_target), {
+// return true if the selector needs to be modified
+EM_JS(void, sapp_js_init, (const char* canvas_selector_cstr), {
+    if (Module.canvas) {
+        specialHTMLTargets["!canvas"] = Module.canvas;
+        stringToUTF8("!canvas", canvas_selector_cstr, 8);
+        return;
+    }
+    
     // lookup and store canvas object by name
-    const target_str = UTF8ToString(c_str_target);
-    Module.sapp_emsc_target = document.getElementById(target_str);
-    if (!Module.sapp_emsc_target) {
-        console.log("sokol_app.h: invalid target:" + target_str);
+    const canvas_selector = UTF8ToString(canvas_selector_cstr);
+    Module.canvas = document.querySelector(canvas_selector);
```

Comment:
Wouldn't this overwrite the standard `Module['canvas'] that might have been set in JS code?

(apart from that `Module.canvas` might cause problems with the Emscripten Closure pass which tends to minify JS property names, that's why the Emscripten JS shims use `Module['canvas']` which is safe from minification.

Reply:
No, it returns first. Seems like magnification also does not effect this. Can you show an example of it not working?

Reply:
It should be noted, the name is semi-arbitrary, but it makes sesne to me to use the name that is used in other things (SDL, glfw, etc.) In js, if I set my selector to `!canvas` and added `Module.sapp_emsc_target` to special-selectors before loading, it would mostly work, except sokol mixes strategies a bit, in different places, using a selector, id, and the dom-reference. This PR unifies that all to point to 1 thing (the canvas, however you set that up.) In places where you use a selector, it uses `!canvas`, in places where it uses an id, it was moved to use selector, and in places where it uses `sapp_emsc_target` it uses `canvas` which all points to the same thing, now, which is the same as the js-side `canvas` option. I think this actually improves things for people who are not using `canvas` option, because it's all 1 strategy.